src/viewport/viewport.service.ts

```diff
@@ -148,24 +99,32 @@ export class ViewportService {
 		if (width === this.lastWidthCheck && this.lastWidthSizeInfo) {
 			return this.lastWidthSizeInfo;
 		}
+
 		this.lastWidthCheck = width;
+		this.lastWidthSizeInfo = this.getWidthSizeInfo(width);
+		return this.lastWidthSizeInfo;
+	}
 
-		// todo: make this more dynamic + decouple from lodash
-		if (_.inRange(width, viewportSizesConfig.xsmall)) {
-			this.lastWidthSizeInfo = viewportSizeRefs[ViewportSizeType.xsmall];
-		} else if (_.inRange(width, viewportSizesConfig.xsmall, viewportSizesConfig.small)) {
-			this.lastWidthSizeInfo = viewportSizeRefs[ViewportSizeType.small];
-		} else if (_.inRange(width, viewportSizesConfig.small, viewportSizesConfig.medium)) {
-			this.lastWidthSizeInfo = viewportSizeRefs[ViewportSizeType.medium];
-		} else if (_.inRange(width, viewportSizesConfig.medium, viewportSizesConfig.large)) {
-			this.lastWidthSizeInfo = viewportSizeRefs[ViewportSizeType.large];
-		} else if (_.inRange(width, viewportSizesConfig.large, viewportSizesConfig.xlarge)) {
-			this.lastWidthSizeInfo = viewportSizeRefs[ViewportSizeType.xlarge];
-		} else if (_.inRange(width, viewportSizesConfig.xlarge, viewportSizesConfig.xxlarge)) {
-			this.lastWidthSizeInfo = viewportSizeRefs[ViewportSizeType.xxlarge];
-		} else {
-			this.lastWidthSizeInfo = viewportSizeRefs[ViewportSizeType.xxlarge1];
+	/**
+	 * Returns the current viewport information
+	 * @param width the viewport width
+	 */
+	private getWidthSizeInfo(width: number): ViewportSizeTypeInfo {
+		// todo: make this more dynamic
+		if (width <= this.config.viewport.breakpoints.xsmall) {
+			return this.viewportDictionary[ViewportSizeType.xsmall];
+		} else if (width <= this.config.viewport.breakpoints.small) {
+			return this.viewportDictionary[ViewportSizeType.small];
+		} else if (width <= this.config.viewport.breakpoints.medium) {
+			return this.viewportDictionary[ViewportSizeType.medium];
+		} else if (width <= this.config.viewport.breakpoints.large) {
+			return this.viewportDictionary[ViewportSizeType.large];
+		} else if (width <= this.config.viewport.breakpoints.xlarge) {
+			return this.viewportDictionary[ViewportSizeType.xlarge];
+		} else if (width <= this.config.viewport.breakpoints.xxlarge) {
+			return this.viewportDictionary[ViewportSizeType.xxlarge];
 		}
-		return this.lastWidthSizeInfo;
+
+		return this.viewportDictionary[ViewportSizeType.xxlarge1];
```

Comment:
extracted into another function to retrieve the width size information.

+ removed usage of lodash